### PR TITLE
[Storage] Added support for https traffic only.

### PR DESCRIPTION
**Pull request by JasonYang-MSFT**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-storage/2016-12-01/swagger/storage.json [hunk 1]
```diff
@@ -652,6 +652,11 @@
             "name": "AccessTier",
             "modelAsString": false
           }
+        },
+        "supportsHttpsTrafficOnly": {
+          "type": "boolean",
+          "default": "false",
```
Comment (vishrutshah) [inline arm-storage/2016-12-01/swagger/storage.json]: Is this "false" a string false expected from service?

File: arm-storage/2016-12-01/swagger/storage.json [hunk 2]
```diff
@@ -940,6 +950,11 @@
...
```
Comment (vishrutshah) [inline arm-storage/2016-12-01/swagger/storage.json]: same as above

**Review by vishrutshah — CHANGES_REQUESTED** [review]
File is not semantically correct. Please review https://travis-ci.org/Azure/azure-rest-api-specs/jobs/222040448

File: arm-storage/2016-12-01/swagger/storage.json (hunk 1 above)
Comment (JasonYang-MSFT) [inline arm-storage/2016-12-01/swagger/storage.json]: Changed to "default": false.

**Comment by vishrutshah** [comment]
@JasonYang-MSFT This changes looks good to me. 

Although there are linter errors from the swagger not related to your changes, I have opened a bug https://github.com/Azure/azure-rest-api-specs/issues/1141 and assign to you or your team to triage and decide. Let us know if you need any more information. Thanks

**Review by vishrutshah — APPROVED** [review]
LGTM!

**Pull request merged**